services/wallet/walletconnect/database.go

```diff
@@ -0,0 +1,76 @@
+package walletconnect
+
+import (
+	"database/sql"
+)
+
+type Pairing struct {
+	Topic        Topic    `json:"topic"`
+	Expiry       int64    `json:"expiry"`
+	Disconnected bool     `json:"disconnected"`
+	AppName      string   `json:"appName"`
+	URL          string   `json:"url"`
+	Description  string   `json:"description"`
+	Icon         string   `json:"icon"`
+	Verified     Verified `json:"verified"`
+}
+
+func InsertPairing(db *sql.DB, pairing Pairing) error {
+	insertSQL := `INSERT INTO wallet_connect_pairings (topic, expiry_timestamp, disconnected, app_name, url, description, icon, verified_is_scam, verified_origin, verified_verify_url, verified_validation) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`
+
+	_, err := db.Exec(insertSQL, pairing.Topic, pairing.Expiry, pairing.Disconnected, pairing.AppName, pairing.URL, pairing.Description, pairing.Icon, pairing.Verified.IsScam, pairing.Verified.Origin, pairing.Verified.VerifyURL, pairing.Verified.Validation)
+	return err
+}
+
+func GetPairingByTopic(db *sql.DB, topic Topic) (*Pairing, error) {
+	querySQL := `SELECT topic, expiry_timestamp, disconnected, app_name, url, description, icon, verified_is_scam, verified_origin, verified_verify_url, verified_validation FROM wallet_connect_pairings WHERE topic = ?`
+
+	row := db.QueryRow(querySQL, topic)
+
+	var pairing Pairing
+	err := row.Scan(&pairing.Topic, &pairing.Expiry, &pairing.Disconnected, &pairing.AppName, &pairing.URL, &pairing.Description, &pairing.Icon, &pairing.Verified.IsScam, &pairing.Verified.Origin, &pairing.Verified.VerifyURL, &pairing.Verified.Validation)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pairing, nil
+}
+
+// GetActivePairings returns all active pairings (not disconnected and not expired) that have an expiry timestamp newer or equal to the given timestamp.
+func GetActivePairings(db *sql.DB, expiryNotOlderThanTimestamp int64) ([]Pairing, error) {
+	querySQL := `SELECT topic, expiry_timestamp, disconnected, app_name, url, description, icon, verified_is_scam, verified_origin, verified_verify_url, verified_validation FROM wallet_connect_pairings WHERE disconnected = 0 AND expiry_timestamp >= ? ORDER BY expiry_timestamp DESC`
+
+	rows, err := db.Query(querySQL, expiryNotOlderThanTimestamp)
+	if err != nil {
+		return nil, err
+	}
+
+	pairings := make([]Pairing, 0)
+	for rows.Next() {
+		var pairing Pairing
+		err := rows.Scan(&pairing.Topic, &pairing.Expiry, &pairing.Disconnected, &pairing.AppName, &pairing.URL, &pairing.Description, &pairing.Icon, &pairing.Verified.IsScam, &pairing.Verified.Origin, &pairing.Verified.VerifyURL, &pairing.Verified.Validation)
+		if err != nil {
+			return nil, err
+		}
+		if err := rows.Err(); err != nil {
+			return nil, err
+		}
+		pairings = append(pairings, pairing)
+	}
+
```

Comment:
@stefandunca we should check here for potential errors from calling `rows.Next()`, like this:
```
if err := rows.Err(); err != nil {
    return nil, err
}
```